### ISSUE TITLE
[WOTLK] Marauding Crust Burster (16857) Doesn't Quite Burst From the Crust

### ISSUE THREAD

**Issue by Vladskywolf** [cmangos/issues#1437]
**Current behavior**: 
Go near a Marauding Crust Burster in Hellfire Peninsula as they're moving around underneath the surface. You can aggro them, they'll spit poison at you, and you can fight them, but they're essentially invisible until they've been killed, at which point they'll become visible. Makes it hard to know which ones are attacking you if you're around several of them.

**Expected behavior**: 
The creature should visibly rise up from underneath the surface and attack you.

**Commit hash**: 
eddeff7fe9ba49ee1dc2ae7df31c0b1fca4bf247

As a workaround I've removed the SD2 script name from the creature_template entry and inserted EventAI scripts from an older build (12576) which works okay.

> DELETE FROM creature_ai_scripts WHERE creature_id = 16857;
>
> INSERT INTO `creature_ai_scripts` (`id`, `creature_id`, `event_type`, `event_inverse_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action1_type`, `action1_param1`, `action1_param2`, `action1_param3`, `action2_type`, `action2_param1`, `action2_param2`, `action2_param3`, `action3_type`, `action3_param1`, `action3_param2`, `action3_param3`, `comment`) VALUES
('1685701','16857','11','0','100','0','0','0','0','0','19','33554432','0','0','0','0','0','0','0','0','0','0','Marauding Crust Burster - Remove Unselectable Unit Flag on Spawn'),
('1685702','16857','11','0','100','0','0','0','0','0','11','29147','0','16','11','33928','0','0','21','0','0','0','Marauding Crust Burster - Cast Sandworm Submerge Visual and Cast Tunnel Bore Passive and Prevent Combat Movement on Spawn'),
('1685703','16857','1','0','100','1','1000','1000','1000','1000','11','29147','0','48','0','0','0','0','0','0','0','0','Marauding Crust Burster - Cast Sandworm Submerge Visual OOC'),
('1685704','16857','4','0','100','0','0','0','0','0','28','0','29147','0','28','0','33928','0','23','1','0','0','Marauding Crust Burster - Remove Sandworm Submerge Visual and Remove Tunnel Bore Passive and Set Phase 1 on Aggro'),
('1685705','16857','9','5','100','1','0','50','2100','4500','11','31747','1','0','0','0','0','0','0','0','0','0','Marauding Crust Burster - Cast Poison (Phase 1)'),
('1685706','16857','9','5','100','1','50','100','500','500','11','29147','0','16','11','33928','0','0','21','1','0','0','Marauding Crust Burster - Cast Sandworm Submerge Visual and Cast Tunnel Bore Passive and Set Phase 2  at 50 Yards (Phase 1)'),
('1685707','16857','9','5','100','1','50','100','500','500','23','1','0','0','0','0','0','0','0','0','0','0','Marauding Crust Burster - Enable Combat Movement at 50 Yards (Phase 1)'),
('1685708','16857','9','3','100','1','0','8','500','500','28','0','29147','0','28','0','33928','0','21','0','0','0','Marauding Crust Burster - Remove Sandworm Submerge Visual and Remove Tunnel Bore Passive and Prevent Combat Movement at 8 Yards (Phase 2)'),
('1685709','16857','9','3','100','1','0','8','500','500','23','-1','0','0','0','0','0','0','0','0','0','0','Marauding Crust Burster - Set Phase 1 at 8 Yards (Phase 2)'),
('1685710','16857','7','0','100','0','0','0','0','0','11','29147','0','21','11','33928','0','0','21','0','0','0','Marauding Crust Burster - Cast Sandworm Submerge Visual and Cast Tunnel Bore Passive and Prevent Combat Movement on Evade'),
('1685711','16857','7','0','100','0','0','0','0','0','22','0','0','0','0','0','0','0','0','0','0','0','Marauding Crust Burster - Set Phase 0 on Evade');
> 
> DELETE FROM creature_template WHERE Entry = 16857;
> 
> INSERT into `creature_template` (`Entry`, `Name`, `SubName`, `IconName`, `MinLevel`, `MaxLevel`, `DifficultyEntry1`, `DifficultyEntry2`, `DifficultyEntry3`, `ModelId1`, `ModelId2`, `ModelId3`, `ModelId4`, `FactionAlliance`, `FactionHorde`, `Scale`, `Family`, `CreatureType`, `InhabitType`, `RegenerateStats`, `RacialLeader`, `NpcFlags`, `UnitFlags`, `DynamicFlags`, `ExtraFlags`, `CreatureTypeFlags`, `SpeedWalk`, `SpeedRun`, `UnitClass`, `Rank`, `Expansion`, `HealthMultiplier`, `PowerMultiplier`, `DamageMultiplier`, `DamageVariance`, `ArmorMultiplier`, `ExperienceMultiplier`, `MinLevelHealth`, `MaxLevelHealth`, `MinLevelMana`, `MaxLevelMana`, `MinMeleeDmg`, `MaxMeleeDmg`, `MinRangedDmg`, `MaxRangedDmg`, `Armor`, `MeleeAttackPower`, `RangedAttackPower`, `MeleeBaseAttackTime`, `RangedBaseAttackTime`, `DamageSchool`, `MinLootGold`, `MaxLootGold`, `LootId`, `PickpocketLootId`, `SkinningLootId`, `KillCredit1`, `KillCredit2`, `QuestItem1`, `QuestItem2`, `QuestItem3`, `QuestItem4`, `QuestItem5`, `QuestItem6`, `MechanicImmuneMask`, `SchoolImmuneMask`, `ResistanceHoly`, `ResistanceFire`, `ResistanceNature`, `ResistanceFrost`, `ResistanceShadow`, `ResistanceArcane`, `PetSpellDataId`, `MovementType`, `MovementTemplateId`, `TrainerType`, `TrainerSpell`, `TrainerClass`, `TrainerRace`, `TrainerTemplateId`, `VendorTemplateId`, `EquipmentTemplateId`, `VehicleTemplateId`, `GossipMenuId`, `AIName`, `ScriptName`) 
values
('16857','Marauding Crust Burster',NULL,NULL,'59','60','0','0','0','16395','0','0','0','14','14','1','0','10','3','3','0','0','33587200','0','0','0','1','1.19048','1','0','1','1','1','1','1','1','1','4142','4979','0','0','104','138','72','106','3750','252','26','2000','0','0','0','0','16857','0','0','0','0','0','0','0','0','0','0','0','0','0','0','0','0','0','0','0','1','0','0','0','0','0','0','0','0','0','0','EventAI','');

[//]: # (If this is a crash report, include the crashlog from a debug build with https://gist.github.com/)

**Comment by killerwife** [comment]
While I understand you wanting to fix it, but this is not the way to go. This is not how it works in sniff. The submerging works through this:

unitTarget->SetStandState(m_spellInfo->Id == 37751 ? UNIT_STAND_STATE_CUSTOM : UNIT_STAND_STATE_STAND);

UNIT_STAND_STATE_CUSTOM  - submerged
UNIT_STAND_STATE_STAND - emerged

I reckon that this spell is failing for some reason. If you really wish to know why it doesn't work as intended, try to see if the STATE_STAND is being given correctly on combat start.

**Comment by Vladskywolf** [comment]
Thanks for this info. I'll take a look at that when i get a chance.

Once thing I did forget to mention in my original post is that when I was flying towards an area where I could test these creatures (before I applied my so-called workaround), I noticed they would pop up out of the ground momentarily without being aggro'd, but from ground level I didn't see this at all. I thought it odd since I don't recall that behavior from when I played TBC live.
